powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/ActivationService.java

```diff
@@ -459,28 +447,23 @@ public ActivationLayer1Response createActivation(ActivationLayer1Request request
                     return prepareEncryptedResponse(response.getEncryptedData(), response.getMac(),
                             response.getNonce(), response.getTimestamp(), processedCustomAttributes, userInfo);
                 }
-                default -> {
-                    logger.warn("Invalid activation request");
-                    throw new PowerAuthInvalidRequestException();
-                }
+                default ->
+                        throw new PowerAuthInvalidRequestException("Unsupported activation type: " + request.getType());
             }
         } catch (PowerAuthClientException ex) {
             if (ex.getPowerAuthError().orElse(null) instanceof final PowerAuthErrorRecovery errorRecovery) {
                 logger.debug("Invalid recovery code, current PUK index: {}", errorRecovery.getCurrentRecoveryPukIndex());
                 throw new PowerAuthRecoveryException(ex.getMessage(), "INVALID_RECOVERY_CODE", errorRecovery.getCurrentRecoveryPukIndex());
             }
-            logger.warn("Creating PowerAuth activation failed, error: {}", ex.getMessage());
-            logger.debug(ex.getMessage(), ex);
-            throw new PowerAuthActivationException();
-        } catch (PowerAuthActivationException ex) {
-            // Do not swallow PowerAuthActivationException for custom activations.
-            // See: https://github.com/wultra/powerauth-restful-integration/issues/199
```

Comment:
The exception is not swallowed but passed as a cause and logged by `PowerAuthExceptionHandler`.